lib/middleware/session.js

```diff
@@ -79,7 +79,14 @@ var ServletSession = exports.ServletSession = function(request) {
                 // to the attributes in the servlet session object.
                 data = new JavaAdapter(org.mozilla.javascript.NativeObject, {
                     put: function(name, start, value) {
-                        getSession().setAttribute(name, value);
+                    	if (typeof(value) === 'undefined' || value === null) {
```

Comment:
+1 drop it. same with the Object.prototype stuff below (which was already there)